runtime/lua/vim/_defaults.lua

```diff
@@ -928,14 +928,17 @@ do
 
   vim.api.nvim_create_autocmd('VimEnter', {
     group = vim.api.nvim_create_augroup('nvim.find_exrc', {}),
-    desc = 'Find project-local configuration',
+    desc = 'Find exrc files in parent directories',
     callback = function()
-      if vim.o.exrc then
-        local files = vim.fs.find(
-          { '.nvim.lua', '.nvimrc', '.exrc' },
-          { type = 'file', upward = true, limit = math.huge }
-        )
-        for _, file in ipairs(files) do
+      local files = vim.fs.find({ '.nvim.lua', '.nvimrc', '.exrc' }, {
+        type = 'file',
+        upward = true,
+        limit = math.huge,
+        -- exrc in cwd already handled from C, thus start in parent directory.
+        path = vim.fs.dirname(vim.uv.cwd()),
+      })
+      for _, file in ipairs(files) do
+        if vim.o.exrc then
```

Comment:
This is a minor thing but I think we should this if check _after_ the loadstring/nvim_exec2, with a comment that explains why we are checking a second time ("If the user cleared 'exrc' then stop searching")